docs/Contributing/Blueprints/Fleet-Installers.md

```diff
@@ -0,0 +1,431 @@
+so that I can add hosts to Fleet without having to know how to successfully generate an
+installer with the `fleetctl package` command.
+
+Figma wireframes: https://www.figma.com/file/hdALBDsrti77QuDNSzLdkx/?node-id=6740%3A267448
+```
+
+## Command `fleetctl package`
+
+Currently users use the `fleetctl package` command to generate Fleet-osquery packages.
+
+The `fleetctl package` command has the following *required* configuration that is specific to a "Fleet Deployment":
+- `--fleet-url`: The URL that the hosts will use to connect to the Fleet server.
+- `--enroll-secret`: Global or team enroll secret to use when enrolling the host to Fleet.
+
+As the goal states, we would like to provide functionality in Fleet to automatically generate and download such packages from the UI.
+
+## How
+
+We can implement such functionality in two ways:
+
+- Option A. Fleet Server to generate such packages itself.
+- Option B. Separate "Packager" service.
+
+There's a lot of platform specific logic and tooling involved in packaging, and one of Fleet's primary goals is to keep deployment/infrastructure simple for On-Prem deployments.
+To that end, we believe the best option is Option B, implementing the functionality as a separate service.
+
+## Packager Service
+
+The "Fleet Packager" service will implement all the package generation logic. Think of it like offering the `fleetctl package` command functionality but as a REST service.
+
+```mermaid
+graph LR;
+    A[User-Agent/<br>Browser]-- Fleet API -->B[Fleet<br>UI/Server]
+    A-- Packager<br>API -->C
+    subgraph FleetDM Hosted Infrastructure
+        direction LR
+        subgraph <br>pkg.fleetctl.com
+            direction TB;
+            C[Packager<br>Service]
+            packages[(Generated<br>packages)]
+        end
+        direction TB;
+        D[tuf.fleetctl.com]    
+        C-- Fetch<br>Targets -->D;
+    end
+    E[Apple's <br> Notary <br> Server]
+    C-- Notary<br>API ---->E
+```
+
+Configurations:
+- The `Fleet UI/Server` will allow configuring the "Fleet Packager" URL (default value being FleetDM Hosted Packager service, something like `pkg.fleetctl.com`).
+- The `Packager Service` will allow configuring an alternative TUF server URL and TUF roots via an environmental variable (default value being FleetDM Hosted TUF, `tuf.fleetctl.com`). 
+
+Both of these configurations will allow users to deploy their own `Packager Service`.
+
+### Storage
+
+The generated packages should be stored on encrypted disk and will expire (will be deleted) after a configurable amount of time (default 30m).
+There are two reasons we want to expire generated packages soon:
+1. To not store user credentials for too long (URL and enroll secret).
+2. To free up space.
+
+We can use "Storage Optimized" instances (see https://aws.amazon.com/ec2/instance-types/).
+
+PS: As a possible future optimization, we could use "Memory Optimized" instances and store packages in RAM instead of using hard disk.
```

Comment:
If I have to bet, this won't be the bottleneck, sounds like transferring the data over the network to the user doing the request might be.

Reply:
Totally agree. First optimizations to try should be on the network, e.g. like caching TUF targets (which do not change often).